test/edu/csus/ecs/pc2/core/imports/LoadICPCTSVDataTest.java

```diff
@@ -111,5 +115,88 @@ private void checkIfGroupsAssigned(List<Account> accountList) {
+    }
+
+    private String getTestSampleContestConfigDirectory(String contestName) {
+        return getTestSampleContestDirectory(contestName) + File.separator + IContestLoader.CONFIG_DIRNAME;
+    }
+
+    private String getTestSampleContestConfigFile(String contestName, String filename) {
+        return getTestSampleContestDirectory(contestName) + File.separator + IContestLoader.CONFIG_DIRNAME + File.separator + filename;
+    }
+
+    /**
+     * Test loading a teams.tsv twice.
+     * 
+     * The 2nd load of teams and groups, in particular groups, caused all accounts' groups to be "empty"
+     * on the Accounts tab.  See 
+     * 
+     * @throws Exception
+     */
+    public void testReLoadTSV() throws Exception {
+
+        String groupsFilename = getTestSampleContestConfigFile("mini", LoadICPCTSVData.GROUPS_FILENAME);
+        assertFileExists(groupsFilename);
+
+        IInternalContest contest = loadSampleContest(null, "mini");
+        assertNotNull(contest);
+        IInternalController controller = new SampleContest().createController(contest, true, false);
+        
+        LoadICPCTSVData loader = new LoadICPCTSVData();
+        loader.setContestAndController(contest, controller);
+        
+        boolean loaded = loader.loadFiles(groupsFilename, false, false);
+        assertTrue("Expecting min contest loaded", loaded);
 
+        assertEquals("Number of groups", 12, contest.getGroups().length);
+        assertEquals("Number of accounts", 167, contest.getAccounts().length);
+
+        loader.checkFiles(groupsFilename);
+
+        /**
+         * Second load of groups from groups.tsv
+         */
+        Group[] loadedGroups = ICPCTSVLoader.loadGroups(loader.getGroupsFilename());
+        Account[] accounts = ICPCTSVLoader.loadAccounts(loader.getTeamsFilename());
+
+        /**
+         * Groups from contest/model, the authoritative groups from model.
+         */
+        Group[] modelGroups = contest.getGroups();
+
+        assertEquals("Number of groups", 12, modelGroups.length);
+        assertEquals("Number of accounts", 151, accounts.length);
+
+        /**
+         * Second load of groups from groups.tsv
+         */
+        // Load groups from file, merge with modelGroups list.
+        Group[] mergedGroups = ICPCTSVLoader.loadGroups(loader.getGroupsFilename(), modelGroups);
+
+        Account[] accounts2 = ICPCTSVLoader.loadAccounts(loader.getTeamsFilename());
+
+        assertEquals("Number of groups", 12, mergedGroups.length);
+        assertEquals("Number of accounts", 151, accounts2.length);
+
+        for (int i = 0; i < modelGroups.length; i++) {
+            assertEquals("(" + i + ") Expecting same group element id for group " + modelGroups[i], modelGroups[i].getElementId(), mergedGroups[i].getElementId());
+        }
+
+        // check that all accounts assigned groups in the model/contest
+        for (Account account : accounts2) {
+            Group group = contest.getGroup(account.getGroupId());
```

Comment:
`account.getGroupId()` can return null (if no Group is assigned).  I don't know if that's a problem here or not...

Reply:
It can, and if it does then this unit test will fail... and the cause can be fixed/researched.